// @pixi/node/lib/index.mjs
import './adapter/index.mjs';
import { extensions, INSTALLED } from '@pixi/core';
export * from '@pixi/core';
export { settings } from '@pixi/core';
import { AlphaFilter } from '@pixi/filter-alpha';
import { BlurFilter, BlurFilterPass } from '@pixi/filter-blur';
import { ColorMatrixFilter } from '@pixi/filter-color-matrix';
import { DisplacementFilter } from '@pixi/filter-displacement';
import { FXAAFilter } from '@pixi/filter-fxaa';
import { NoiseFilter } from '@pixi/filter-noise';
import '@pixi/mixin-cache-as-bitmap';
import '@pixi/mixin-get-child-by-name';
import '@pixi/mixin-get-global-position';
import { loadTextures, loadWebFont } from '@pixi/assets';
export * from '@pixi/assets';
import { ResizePlugin } from '@pixi/app';
export * from '@pixi/app';
import { loadBitmapFont } from '@pixi/text-bitmap';
export * from '@pixi/text-bitmap';
export * from '@pixi/display';
export * from '@pixi/extract';
export * from '@pixi/graphics';
export * from '@pixi/mesh';
export * from '@pixi/mesh-extras';
export * from '@pixi/particle-container';
export * from '@pixi/prepare';
export * from '@pixi/sprite';
export * from '@pixi/sprite-animated';
export * from '@pixi/sprite-tiling';
export * from '@pixi/spritesheet';
export * from '@pixi/text';
import { NodeCanvasResource } from './adapter/NodeCanvasResource.mjs';
export { NodeCanvasResource } from './adapter/NodeCanvasResource.mjs';
export { NodeCanvasElement } from './adapter/NodeCanvasElement.mjs';
export { loadNodeBitmapFont } from './adapter/loadNodeBitmapFont.mjs';
export { loadNodeFont } from './adapter/loadNodeFont.mjs';
export { loadNodeTexture } from './adapter/loadNodeTexture.mjs';
export { loadNodeBase64 } from './adapter/loadNodeBase64.mjs';
export { NodeAdapter } from './adapter/adapter.mjs';

extensions.remove(loadTextures, loadWebFont, loadBitmapFont, ResizePlugin);
INSTALLED.length = 0;
INSTALLED.push(NodeCanvasResource);
const filters = {
  AlphaFilter,
  BlurFilter,
  BlurFilterPass,
  ColorMatrixFilter,
  DisplacementFilter,
  FXAAFilter,
  NoiseFilter
};

export { filters };
